server/neptune/gateway/event/comment_handler.go

```diff
@@ -44,77 +49,171 @@ type Comment struct {
+		scheduler:            scheduler,
+		commentCreator:       commentCreator,
+		rootDeployer:         rootDeployer,
+		vcsStatusUpdater:     vcsStatusUpdater,
+		globalCfg:            globalCfg,
+		projectCommandGetter: projectCommandGetter,
 	}
 }
 
 type CommentEventWorkerProxy struct {
-	logger           logging.Logger
-	snsWriter        Writer
-	allocator        feature.Allocator
-	scheduler        scheduler
-	commentCreator   commentCreator
-	rootDeployer     rootDeployer
-	vcsStatusUpdater statusUpdater
-	globalCfg        valid.GlobalCfg
+	logger               logging.Logger
+	scope                tally.Scope
+	snsWriter            Writer
+	allocator            feature.Allocator
+	scheduler            scheduler
+	commentCreator       commentCreator
+	rootDeployer         rootDeployer
+	vcsStatusUpdater     statusUpdater
+	globalCfg            valid.GlobalCfg
+	projectCommandGetter projectCommandGetter
 }
 
 func (p *CommentEventWorkerProxy) Handle(ctx context.Context, request *http.BufferedRequest, event Comment, cmd *command.Comment) error {
 	shouldAllocate, err := p.allocator.ShouldAllocate(feature.PlatformMode, feature.FeatureContext{
 		RepoName: event.BaseRepo.FullName,
 	})
 
+	// typically we shouldn't be failing if we can't figure out the feature, however, there is some complex logic
+	// that depends on us knowing which mode we are running on so in order to prevent unintended consequences, let's just
+	// bail if this happens.
 	if err != nil {
 		p.logger.ErrorContext(ctx, "unable to allocate platform mode")
-		return p.handleLegacyComment(ctx, request, event, cmd, shouldAllocate)
+		return errors.Wrap(err, "unable to allocate platform mode feature")
+	}
+
+	if !shouldAllocate {
+		return p.handleLegacyComment(ctx, request, event, cmd)
+	}
+
+	cmdCtx := &command.Context{
+		HeadRepo:   event.HeadRepo,
+		Pull:       event.Pull,
+		Scope:      p.scope,
+		User:       event.User,
+		Log:        p.logger,
+		Trigger:    command.CommentTrigger,
+		RequestCtx: ctx,
+	}
+
+	cmds, err := p.projectCommandGetter.GetProjectCommands(cmdCtx, event.BaseRepo, cmdCtx.Pull)
+	if err != nil {
+		if err != nil {
+			if _, statusErr := p.vcsStatusUpdater.UpdateCombined(ctx, cmdCtx.HeadRepo, cmdCtx.Pull, models.FailedVCSStatus, cmd.Name, "", err.Error()); statusErr != nil {
+				cmdCtx.Log.WarnContext(cmdCtx.RequestCtx, fmt.Sprintf("unable to update commit status: %v", statusErr))
+			}
+			return errors.Wrap(err, "getting project commands")
+		}
+	}
+
+	platformModeRoots, defaultModeRoots := partitionRootsByMode(cmds)
+	p.notifyImpendingChanges(
+		ctx,
+		len(platformModeRoots) == len(cmds),
+		request,
+		event,
+		cmd,
+	)
+
+	if !cmd.ForceApply {
+		return p.forwardToSns(ctx, request)
+	}
+
+	// first process any platform mode force applies.
+	var rootNames []string
+	for _, r := range platformModeRoots {
+		rootNames = append(rootNames, r.ProjectName)
+	}
+
+	if err := p.scheduler.Schedule(ctx, func(ctx context.Context) error {
```

Comment:
I think the entire handle function should be wrapped in the scheduler since `GetProjectCommands` should be performing a git clone.

Reply:
good call, will do that.